test/functional/feature_staking_free_tx.py

```diff
@@ -56,26 +56,59 @@ def free_tx_simple_test(self):
         node1_height = self.nodes[1].getblockcount()
         assert node0_height == node1_height == (starting_height + 10), 'Difference in nodes height'
 
-        # send staking burn transaction
         stake_id = self.send_staking_deposit_tx(addr2, deposit_value, 0)
         free_tx_base_id = self.nodes[0].sendtoaddress(addr2, free_tx_base_value)
+        free_tx_base_id2 = self.nodes[0].sendtoaddress(addr2, free_tx_base_value)
         invalid_free_tx_base_id = self.nodes[0].sendtoaddress(addr1, free_tx_base_value)
 
         self.nodes[0].generate(1)
         self.sync_all()
-        # check if the transaction over the hard-coded limit of 1kB will be rejected
-        assert_raises_rpc_error(-26, "invalid-free-transaction", self.send_free_tx, dummy_addresses, free_tx_amount, 0, addr2)
+        # check if the transaction over the limit will be rejected
+        assert_raises_rpc_error(-26, "invalid-free-transaction", self.send_free_tx, dummy_addresses, free_tx_amount, 0,
+                                addr2)
 
         # check if the transaction that has no underlying stake will be rejected
-        assert_raises_rpc_error(-26, "invalid-free-transaction", self.send_free_tx, [self.nodes[1].getnewaddress()], free_tx_amount, 0, addr1)
+        assert_raises_rpc_error(-26, "invalid-free-transaction", self.send_free_tx, [self.nodes[1].getnewaddress()],
+                                free_tx_amount, 0, addr1)
```

Comment:
If we use more verbose messages we will have easier time explaining it to the FE

Reply:
There are many things that could go wrong with our free transaction and finding the exact cause requires building quite a big logic, which is not the purpose of the node, and which is reflected in the error convention - they are all rather laconic. The convention puts the user as the one responsible for finding the error, not the node.

Reply:
My main issue is that we have two "invalid-free-transaction" messages in logs meanwhile we have two different comments to each of them. So to make error verification more easy we should at least name the second one: "invalid-free-transaction-stage-2"